tcp-keepalive/internal/server/server.go

```diff
@@ -26,18 +29,31 @@ func initLogger() {
 	log = slog.New(slog.NewJSONHandler(os.Stdout, nil))
 }
 
-func NewServer(cfg *Config) (*Server, error) {
+func NewServer(cfg *Config, mcfg *metrics.Config) (*Server, error) {
 	if err := cfg.Validate(); err != nil {
 		return nil, err
 	}
 
-	log.Info("new server", slog.Any("config", cfg))
-	return &Server{Config: cfg}, nil
+	m, err := NewMetrics(mcfg)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Info("new server", slog.Any("config", cfg), slog.Any("metrics", m))
+	return &Server{Config: cfg, metrics: m}, nil
 }
 
 func (s *Server) Run(ctx context.Context) error {
 	log = log.With("listen", s.ListenAddr)
 
+	if s.metrics.Export {
+		go func() {
+			for {
```

Comment:
Do we need this loop?

Reply:
@terassyi 
Without this loop, tcp-keepalive will run but won't export any metrics when the metrics server fail with an error.